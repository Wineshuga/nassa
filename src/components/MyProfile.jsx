import { useSelector } from 'react-redux';

const MyProfile = () => {
  const { missions } = useSelector((store) => store.missions);
  return (
    <div className="container d-flex">
      <section className="">
        <h3>My Missions</h3>
        <ul className="list-group">
          {missions.filter((item) => item.reserved)
            .map((item) => (
              <li
                key={item.id}
                className="list-group-item p-4"
              >
                {item.name}
              </li>
            ))}
        </ul>
      </section>
    </div>
  );
};

export default MyProfile;
