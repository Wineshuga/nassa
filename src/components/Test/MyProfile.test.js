import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom/extend-expect';
import MyProfile from '../MyProfile';

const mockStore = configureStore([]);

describe('MyProfile Component', () => {
  it('renders My Missions section correctly', () => {
    const missions = [
      {
        id: 1,
        name: 'Mission 1',
        reserved: true,
      },
    ];

    const store = mockStore({
      missions: { missions },
      rockets: { RocketList: [] },
    });

    const { asFragment, getByText } = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );

    expect(getByText('My Missions')).toBeInTheDocument();
    missions.forEach((mission) => {
      expect(getByText(mission.name)).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders My Rockets section correctly', () => {
    const rockets = {
      RocketList: [
        {
          id: 1,
          name: 'Rocket 1',
          reserved: true,
        },
      ],
    };

    const store = mockStore({
      missions: { missions: [] },
      rockets,
    });

    const { asFragment, getByText } = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );

    expect(getByText('My Rockets')).toBeInTheDocument();
    rockets.RocketList.forEach((rocket) => {
      expect(getByText(rocket.name)).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
