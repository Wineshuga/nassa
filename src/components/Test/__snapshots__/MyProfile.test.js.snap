// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile Component renders My Missions section correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3>
          My Missions
        </h3>
        <ol
          class="list-group"
        >
          <li
            class="list-group-item"
          >
            Mission 1
          </li>
        </ol>
      </div>
      <div
        class="col"
      >
        <h3>
          My Rockets
        </h3>
        You have not reserved any rockets.
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MyProfile Component renders My Rockets section correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3>
          My Missions
        </h3>
        <p>
          You have not reserved any missions.
        </p>
      </div>
      <div
        class="col"
      >
        <h3>
          My Rockets
        </h3>
        <ol
          class="list-group"
        >
          <li
            class="list-group-item"
          >
            Rocket 1
          </li>
        </ol>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
